File: DSA/Aditi-Sharma/Day10/day10_question2.java
import java.util.Stack;
import java.util.*;
class DAY10_S2
{
    private Stack<Integer> stack1;
    private Stack<Integer> stack2;

    public DAY10_S2() 
    {
        stack1 = new Stack<>();
        stack2 = new Stack<>();
    }
    public void enqueue(int item) 
    {
        stack1.push(item);
    }
    public int dequeue() 
    {
        if (stack2.isEmpty()) 
        {
            while (!stack1.isEmpty()) 
            {
                stack2.push(stack1.pop());
            }
        }
        
        if (stack2.isEmpty()) 
        {
            throw new RuntimeException("Queue is empty");
        }
        
        return stack2.pop();
    }

    public static void main(String[] args) 
    {
        DAY10_S2 queue = new DAY10_S2();
        queue.enqueue(1);
        queue.enqueue(2);
        System.out.println(queue.dequeue()); 
        System.out.println(queue.dequeue()); 
    }
}

File: DSA/Vikash_kumar/Day_10_Question_2.py
Day10-Question2-
class QueueUsingStacks:
    def __init__(self):
        self.stack_in = []
        self.stack_out = []

    def enqueue(self, item):
        self.stack_in.append(item)
        print(f"Enqueued {item}")

    def dequeue(self):
        if not self.stack_out:
            while self.stack_in:
                self.stack_out.append(self.stack_in.pop())
        if not self.stack_out:
            print("Queue is empty")
            return None
        return self.stack_out.pop()

queue = QueueUsingStacks()

while True:
    command = input("Enter a command (enqueue <value>, dequeue, or exit): ").strip().lower()
    if command.startswith("enqueue"):
        try:
            _, value = command.split()
            value = int(value)
            queue.enqueue(value)
        except ValueError:
            print("Please provide a valid integer value to enqueue.")
    elif command == "dequeue":
        result = queue.dequeue()
        if result is not None:
            print(f"Dequeued {result}")
    elif command == "exit":
        break
    else:
        print("Invalid command. Please enter 'enqueue <value>', 'dequeue', or 'exit'.")
